File: climb-log-flutter/lib/climasys/api/models/get_gym_grades_by_gym_name_response.dart
import 'package:climasys/climasys/api/models/grade.dart';

class GetGymGradesByGymNameResponse {
  final List<Grade> grades;
  final List<Grade> standardGrades;

  GetGymGradesByGymNameResponse({required this.grades, required this.standardGrades});

  factory GetGymGradesByGymNameResponse.fromJson(Map<String, dynamic> json) {
    return GetGymGradesByGymNameResponse(
        grades: (json['grades'] as List<dynamic>)
          .map((e) => Grade.fromJson(e))
          .toList(),
        standardGrades: (json['standardGrades'] as List<dynamic>)
          .map((e) => Grade.fromJson(e))
          .toList(),
    );
  }
}

File: ClimbLog.Api/ClimbLog.Application/Features/Grades/Queries/GetGymGradesByGymName/GetGymGradesByGymNameResponse.cs
using ClimbLog.Domain.Models.Entities;

namespace ClimbLog.Application.Features.Grades.Queries.GetGymGradesByGymName;

public class GetGymGradesByGymNameResponse
{
    public required List<Grade> Grades { get; set; }
    public required List<Grade> StandardGrades { get; set; }
}
